packages/openactive-integration-tests/test/features/access/access-code/implemented/manual-access-codes-test.js

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable no-unused-vars */
+const chakram = require('chakram');
+const chai = require('chai'); // The latest version for new features than chakram includes
+const { FeatureHelper } = require('../../../../helpers/feature-helper');
+const { GetMatch, C1, C2, B } = require('../../../../shared-behaviours');
+
+/* eslint-enable no-unused-vars */
+
+FeatureHelper.describeFeature(module, {
+  testCategory: 'access',
+  testFeature: 'access-code',
+  testFeatureImplemented: true,
+  testIdentifier: 'manual-access-codes',
+  testName: 'Successful booking with access codes.',
+  testDescription: 'Access codes returned for B request.',
+  // The primary opportunity criteria to use for the primary OrderItem under test
+  testOpportunityCriteria: 'TestOpportunityBookable',
+  // The secondary opportunity criteria to use for multiple OrderItem tests
+  controlOpportunityCriteria: 'TestOpportunityBookable',
+},
+function (configuration, orderItemCriteria, featureIsImplemented, logger, state, flow) {
+  beforeAll(async function () {
+    await state.fetchOpportunities(orderItemCriteria);
+
+    return chakram.wait();
+  });
+
+  afterAll(async function () {
+    await state.deleteOrder();
+    return chakram.wait();
+  });
+
+  describe('Get Opportunity Feed Items', function () {
+    (new GetMatch({
+      state, flow, logger, orderItemCriteria,
+    }))
+      .beforeSetup()
+      .successChecks()
+      .validationTests();
+  });
+
+  describe('C1', function () {
+    (new C1({
+      state, flow, logger,
+    }))
+      .beforeSetup()
+      .successChecks()
+      .validationTests();
+  });
+
+  describe('C2', function () {
+    (new C2({
+      state, flow, logger,
+    }))
+      .beforeSetup()
+      .successChecks()
+      .validationTests();
+  });
+
+  describe('B', function () {
+    (new B({
+      state, flow, logger,
+    }))
+      .beforeSetup()
+      .successChecks()
+      .validationTests();
+
+    it('Response should include accessCode array with appropriate fields (name and description) for each OrderItem', () => {
+      chakram.expect(state.bResponse.body.orderedItem).to.be.an('array');
+
+      state.bResponse.body.orderedItem.forEach((orderItem, orderItemIndex) => {
+        chakram.expect(orderItem.accessCode).to.be.an('array');
+
+        orderItem.accessCode.forEach((accessCode, accessCodeIndex) => {
+          chakram.expect(state.bResponse).to.have.schema(`orderedItem[${orderItemIndex}].accessCode[${accessCodeIndex}].name`, {
+            type: 'string',
+          });
+          chakram.expect(state.bResponse).to.have.schema(`orderedItem[${orderItemIndex}].accessCode[${accessCodeIndex}].description`, {
+            type: 'string',
+          });
```

Comment:
Sorry, this is a weird one, but we try to use chai's expect function as much as possible - using chakram's expect' only for .to.have.status(..)`, which is useful

chakram is based off of old chai code, which it added some HTTP request stuff to. Therefore, we prefer chai, which is up-to-date and well documented.

(I personally think we should bin chakram at some point for something like axios. It's made for a pre-promise JS world (though it does have promise support now) and is not typed or well documented)

chai doesn't have `to.have.schema()` but since it's just checking that those values are strings, you can use `be.a('string')` e.g.

```
chai.expect(state.bResponse).to.have.nested.property(`orderedItem[${orderItemIndex}].accessCode[${accessCodeIndex}].name`)
  .that.is.a('string');
```

Reply:
Ok sure, I didn't know that, 
Nick suggested to use to.have.schema instead of checking the actual value, but your suggestion makes sense, that way we can still check 'structure' without actual values, will change it to chai and that.is.a - string

Reply:
Apologies @NemanjaStefanovic for misleading here!

Reply:
np, I learnt something new :)